e2e/actions/actions.go

```diff
@@ -1723,6 +1724,12 @@ func (c actionTests) fuseMount(t *testing.T) {
 			t,
 			e2e.AsSubtest(tt.name),
 			e2e.WithProfile(tt.profile),
+			e2e.PreRun(func(t *testing.T) {
+				// Host tests require fuse v3 - will fail on host with libfuse v2
```

Comment:
I'm not quite understanding the fix, e2e uses a wrapped sshfs binary from a fresh built image with fuse v3 support, since Singularity use the pre-mount feature of Fuse, it means that Fuse daemon are not calling `fusermount` at all. Those tests are passing on centos6.7 which is far behind Ubuntu 16.04 in term of software and kernel, so I think the issue is elsewhere.

Reply:
Hi @cclerget - I understand that `fusermount` isn't called - was using it as a proxy to check for v3 libfuse on the host - which I understood was required for host mounts.

Maybe there is some kind of library mixing issue here with the sshfs binary that is being called finding the old host library rather than whatever it's meant to use? When you run the tests passing on RHEL6 is the `fuse-libs` package installed on the host at all? I would only get the fuse tests on RHEL6 running at all if I manually `modprobe fuse` as otherwise the require fs fuse fails and they skip.

I am no longer testing on RHEL6 as it has gone end of life (yesterday), and my developer subscription now longer attaches.

Reply:
Here's the actual failure...

```
=== RUN   TestE2E/PAR/ACTIONS/fuse_mount/HostAsUser
    TestE2E/PAR/ACTIONS/fuse_mount/HostAsUser: actions.go:1722: Running command "/usr/local/bin/singularity exec --fusemount host:/tmp/stest.993301134/sshfs-314568944/sshfs-wrapper root@127.0.0.1:/ -p 2022 -F /tmp/stest.993301134/sshfs-314568944/etc/ssh/ssh_config -o IdentityFile=/tmp/stest.993301134/sshfs-314568944/user.key -o StrictHostKeyChecking=no /mnt /tmp/stest.993301134/sshfs-314568944 test -d /mnt/etc"
    TestE2E/PAR/ACTIONS/fuse_mount/HostAsUser: actions.go:1722: 
        "/usr/local/bin/singularity exec --fusemount host:/tmp/stest.993301134/sshfs-314568944/sshfs-wrapper root@127.0.0.1:/ -p 2022 -F /tmp/stest.993301134/sshfs-314568944/etc/ssh/ssh_config -o IdentityFile=/tmp/stest.993301134/sshfs-314568944/user.key -o StrictHostKeyChecking=no /mnt /tmp/stest.993301134/sshfs-314568944 test -d /mnt/etc" output:
        Failed to add the host to the list of known hosts (/home/dave/.ssh/known_hosts).
        fuse: reading device: I/O error
        
    TestE2E/PAR/ACTIONS/fuse_mount/HostAsUser: actions.go:1722: got 1 as exit code and was expecting 0: exit status 1
        waiting for command "/usr/local/bin/singularity exec --fusemount host:/tmp/stest.993301134/sshfs-314568944/sshfs-wrapper root@127.0.0.1:/ -p 2022 -F /tmp/stest.993301134/sshfs-314568944/etc/ssh/ssh_config -o IdentityFile=/tmp/stest.993301134/sshfs-314568944/user.key -o StrictHostKeyChecking=no /mnt /tmp/stest.993301134/sshfs-314568944 test -d /mnt/etc"
        github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func1
        	github.com/sylabs/singularity/e2e/internal/e2e/singularitycmd.go:658
        testing.tRunner
        	testing/testing.go:991
        runtime.goexit
        	runtime/asm_amd64.s:1373
```

Reply:
fusermount3 and libfuse3 are in two different packages (fuse3 vs fuse3-libs on Red Hat); you can't assume that a lack of fusermount3 implies a lack of libfuse3.